nntrainer/graph/network_graph.cpp

```diff
@@ -398,6 +398,8 @@ void NetworkGraph::setBatchSize(unsigned int batch_size) {
                                        ts_grad.getDim().batch());
         }
       }
+      /// override setting batch as per request
+      (*iter)->setBatch(batch_size);
```

Comment:
Are you sure to move it here? If this is moved here, then the changes requested here by the layers will not be reflected (which is done from line 392 to 399)

Reply:
Yeah, this is intended. 
for example below should be override default behavior.
```cpp
  context.updateTensor(wt_idx[RNNCellParams::hidden_state],
                       batch * max_timestep);
```

Please correct me if I got something wrong :)

Reply:
You are right that context.updateTensor() should decide how to update the tensor shapes, but IMU this updated information should reach tensor_pool, which is done from L392 to L399.

Reply:
The memory is always de-allocated with setBatch(), so `context.updateTensor()` alone is sufficient.
L392 to L399 can be removed IMO as its purpose was to pass the information to the tensor_pool if not deallocated, please check.

Reply:
Yeah, the function guarantee's it is deallocated so it should be fine. But I'll delay removing the part as we don't have a proper test scenario that batch is being changed and run train or inference. (this will have to be added to a model test scenario). I am suspecting some layers requires setBatch() but did not create one, I'll leave an issue instead for now.

Reply:
We do have a model test with ccapi :) - https://github.com/nnstreamer/nntrainer/blob/c09bbf73ae234e9c3bd989c59d3fb7af220679b7/test/ccapi/unittest_ccapi.cpp#L364-L386

Reply:
Awesome :)